format-common/src/main/java/io/cdap/plugin/format/input/PathTrackingInputFormatProvider.java

```diff
@@ -63,6 +65,26 @@ protected void validate() {
     // no-op
   }
 
+  protected Schema addPathField(Schema dataSchema, FailureCollector failureCollector) {
+    List<Schema.Field> fields = new ArrayList<>(dataSchema.getFields());
+    addPathField(fields, failureCollector);
+    return Schema.recordOf(dataSchema.getRecordName(), fields);
+  }
+
+  protected void addPathField(List<Schema.Field> dataFields, FailureCollector failureCollector) {
+    String pathField = conf.getPathField();
+    if (pathField != null) {
+      for (Schema.Field field : dataFields) {
+        if (pathField.equals(field.getName())) {
+          failureCollector.addFailure("Pathfield value already exists in the schema",
+            String.format("Please provide a new value instead of %s", pathField));
+          throw failureCollector.getOrThrowException();
```

Comment:
don't throw an exception here. The point of the FailureCollector is to be able to gather multiple problems at once instead of throwing an exception at the first problem.

Reply:
Removed Exception but returned original schema